File: Codes/1780-check-if-number-is-a-sum-of-powers-of-three_20250814.java
/*
 * @Author: LetMeFly
 * @Date: 2025-08-14 10:28:59
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-08-14 18:44:38
 */
class Solution {
    public boolean checkPowersOfThree(int n) {
        while (n > 0) {
            if (n % 3 == 2) {
                return false;
            }
            n /= 3;
        }
        return true;
    }
}

File: Codes/1780-check-if-number-is-a-sum-of-powers-of-three_20250814.py
'''
Author: LetMeFly
Date: 2025-08-14 10:28:59
LastEditors: LetMeFly.xyz
LastEditTime: 2025-08-14 18:43:00
'''
class Solution:
    def checkPowersOfThree(self, n: int) -> bool:
        while n:
            if n % 3 == 2:
                return False
            n //= 3
        return True